craft_parts/executor/part_handler.py

```diff
@@ -332,6 +458,71 @@ def _remove(filename: Path) -> None:
         shutil.rmtree(filename)
 
 
+def _clean_shared_area(
+    *, part_name: str, shared_dir: Path, part_states: Dict[str, StepState]
+) -> None:
+    """Clean files added by a part to a shared directory.
+
+    :param part_name: The name of the part that added the files.
+    :param shared_dir: The shared directory to remove files from.
+    :param part_states: A dictionary containing the each part's state for the
+        step being processed.
+    """
+    # no state defined for this part, we won't remove files
+    if part_name not in part_states:
+        return
+
+    state = part_states[part_name]
+    files = state.files
+    directories = state.directories
+
+    # We want to make sure we don't remove a file or directory that's
+    # being used by another part. So we'll examine the state for all parts
+    # in the project and leave any files or directories found to be in
+    # common.
+    for other_name, other_state in part_states.items():
+        if other_state and (other_name != part_name):
+            files -= other_state.files
+            directories -= other_state.directories
+
+    # Finally, clean the files and directories that are specific to this
+    # part.
+    _clean_migrated_files(files, directories, shared_dir)
+
+
+def _clean_migrated_files(files: Set[str], dirs: Set[str], directory: Path) -> None:
+    """Remove files and directories migrated from part install to a common directory.
+
+    :param files: A set of files to remove.
+    :param dirs: A set of directories to remove.
+    :param directory: The path to remove files and directories from.
+    """
+    for each_file in files:
+        try:
+            os.remove(os.path.join(directory, each_file))
```

Comment:
Other parts of this code work with "pathlib", shouldn't we standarize everything to it?

Reply:
Yes. Fixing it, and there's a task already in place to standardize it across the whole project.